cmd/axelard/cmd/vald/broadcaster/broadcast.go

```diff
@@ -200,3 +216,22 @@ func NewPipelineWithRetry(cap int, maxRetries int, backOffStrategy utils.BackOff
 
 	return p
 }
+
+type causer interface {
+	Cause() error
+}
+
+func isABCIError(err error) bool {
+	for {
+		switch e := err.(type) {
+		case nil:
+			return false
+		case *sdkerrors.Error:
+			return true
+		case causer:
+			err = e.Cause()
```

Comment:
I don't understand this part. Shouldn't the build be broken because we do not return a value if the switch enters this clause?

Reply:
there is a for loop, cause unwraps the error